packages/rum-core/src/performance-monitoring/performance-monitoring.js

```diff
@@ -262,74 +353,50 @@ class PerformanceMonitoring {
       return false
     }
 
-    if (duration > transactionDurationThreshold) {
-      if (__DEV__) {
-        this._logginService.debug(
-          `transaction(${tr.id}, ${tr.name}) was discarded! Transaction duration (${duration}) is greater than the transactionDurationThreshold configuration (${transactionDurationThreshold})`
-        )
-      }
-      return false
-    }
-
-    if (tr.options.managed && tr.spans.length === 0) {
-      if (__DEV__) {
-        this._logginService.debug(
-          `transaction(${tr.id}, ${tr.name}) was discarded! Transaction does not have any spans`
-        )
+    if (tr.isManaged()) {
+      if (duration > TRANSACTION_DURATION_THRESHOLD) {
+        if (__DEV__) {
+          this._logginService.debug(
+            `transaction(${tr.id}, ${tr.name}) was discarded! Transaction duration (${duration}) is greater than managed transaction threshold (${TRANSACTION_DURATION_THRESHOLD})`
+          )
+        }
+        return false
       }
-      return false
-    }
-
-    /**
-     * In case of unsampled transaction, send only the transaction to apm server
-     *  without any spans to reduce the payload size
-     */
-    if (!tr.sampled) {
-      tr.resetSpans()
-    }
-
-    if (tr.options.checkBrowserResponsiveness) {
-      const wasBrowserResponsive = this.checkBrowserResponsiveness(
-        tr,
-        BROWSER_RESPONSIVENESS_INTERVAL,
-        BROWSER_RESPONSIVENESS_BUFFER
-      )
 
-      if (!wasBrowserResponsive) {
+      if (tr.spans.length === 0) {
```

Comment:
Since you moved the `transaction.resetSpans()` inside `adjustTransactionSpans` this check here will prevent upsampled transactions to be sent, we should apply this check only to sampled transactions. We probably need a test as well

Reply:
Nice catch, will fix it and add test as well.